components/pricing_modal/content.tsx

```diff
@@ -241,9 +265,31 @@ function Content(props: ContentProps) {
                         }}
                         planExtraInformation={<StarterDisclaimerCTA/>}
                         buttonDetails={{
-                            action: () => {}, // noop until we support downgrade
-                            text: formatMessage({id: 'pricing_modal.btn.upgrade', defaultMessage: 'Upgrade'}),
-                            disabled: true, // disabled until we have functionality to downgrade
+                            action: () => {
+                                if (usage.teams.active > 1) {
+                                    if (!starterProduct) {
+                                        return;
+                                    }
+                                    dispatch(
+                                        openModal({
+                                            modalId: ModalIdentifiers.CLOUD_DOWNGRADE_CHOOSE_TEAM,
+                                            dialogType: DownGradeTeamRemovalModal,
+                                            dialogProps: {
+                                                product_id: starterProduct?.id,
+                                            },
+                                        }),
+                                    );
+                                } else {
```

Comment:
Looks like the designs also call for a third state, where we open a variation of the limits modal if there is only one team, but one or more of the limits is exceeded: https://www.figma.com/file/Aob5ffHtfjKzQoebR93ycA/Cloud-Freemium?node-id=657%3A436901

![one-team-and-exceeded-limit](https://user-images.githubusercontent.com/13738432/172929405-b0b58f9a-807a-4f8d-80e3-99d170708932.jpg)

It would need to make use of the hardcoded `fallbackStarterLimits` and `fallbackProfessionalLimits` until https://mattermost.atlassian.net/browse/MM-44787 is implemented.

Reply:
The ticket only covers the team downgrade: https://mattermost.atlassian.net/browse/MM-44315

I think because of what you mentioned, but also, we're not actually enforcing the archiving of anything else for the time being (AFAIK)

Reply:
sounds good to me!